src/Workspaces/CSharp/Portable/Execution/CSharpLanguageSpecificSerializationService.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Execution;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Execution
+{
+    [ExportLanguageService(typeof(ILanguageSpecificSerializationService), LanguageNames.CSharp), Shared]
+    internal class CSharpLanguageSpecificSerializationService : AbstractLanguageSpecificSerializationService
+    {
+        public override void WriteTo(CompilationOptions options, ObjectWriter writer, CancellationToken cancellationToken)
```

Comment:
I have review comment somewhere. that would be nice and that would force it to have better way for those resolver's. which I am not implementing right now.

I can open a bug to compiler team.